src/accelerate/utils/modeling.py

```diff
@@ -688,7 +692,10 @@ def infer_auto_device_map(
         devices.append("disk")
 
     # Devices that need to keep space for a potential offloaded layer.
-    main_devices = [gpus[0], "cpu"] if len(gpus) > 0 else ["cpu"]
+    if "mps" in gpus:
+        main_devices = ["mps"]
+    else:
+        main_devices = [gpus[0], "cpu"] if len(gpus) > 0 else ["cpu"]
```

Comment:
Let's not mix and match syntaxes
```suggestion
    if "mps" in gpus:
        main_devices = ["mps"]
    elif len(gpus) > 0:
        main_devices = [gpus[0], "cpu"] if len(gpus) > 0
    else:
        main_devices = ["cpu"]
```